book/src/csi-driver-object.md

```diff
@@ -70,6 +75,19 @@ There are four important fields:
     The default is `Persistent`, which is the normal PVC/PV mechanism. `Ephemeral` enables
     [inline ephemeral volumes](ephemeral-local-volumes.md) in addition (when both
     are listed) or instead of normal volumes (when it is the only entry in the list).
+* `tokenRequests`
+  * This field was added in Kubernetes 1.20 and cannot be set when using an older Kubernetes release.
+  * This field is alpha.
+  * If this field is specified, Kubelet will plumb down the bound service acocunt tokens of the pod as `volume_context` in the `NodePublishVolume`:
+    * `"csi.storage.k8s.io/serviceAccount.tokens": {"gcp":{"token":"<token>","expirationTimestamp":"<expiration timestamp in RFC3339>"}}`
+    * If CSI driver doesn't find token recorded in the `volume_context`, it should return error in `NodePublishVolume` to inform Kubelet to retry.
+    * Audiences should be distinct, otherwise the validation will fail. If the audience is "", it means the issued token has the same audience as kube-apiserver.
```

Comment:
Is there some general documentation on tokens and audiences that we can point to from here? It may be good to specify what kinds of drivers/use cases this would be used for

Reply:
Actually this kind of information might be useful under a feature page: https://kubernetes-csi.github.io/docs/features.html, rather than hidden inside this object page, wdyt?

Reply:
yes, https://tools.ietf.org/html/rfc7519#section-4.1.3. i'll include this in the feature page.